fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -172,7 +174,31 @@ def _call_oracle(self, lambda_vec):
             estimator = clone(estimator=self.estimator, safe=False)
 
         oracle_call_start_time = time()
-        estimator.fit(self.constraints.X, redY, **{self.sample_weight_name: redW})
+
+        # The line that is being replaced
+        # estimator.fit(self.constraints.X, redY, **{self.sample_weight_name: redW})
+
+        # ------- change made starting from here -------- #
+
+        # historical version: try and except, downside: too expensive
+        # try:
+        #     estimator.fit(self.constraints.X, redY, **{self.sample_weight_name: redW})
+        # except TypeError:
+        #     resampled_X, resampled_Y = self.resampling_method(self.constraints.X, redY, redW)
+        #     estimator.fit(resampled_X, resampled_Y)
+
+        # proposing version: if check args
+        if estimator.fit() and 'sample_weight' in inspect.getfullargspec(estimator.fit).args:
+            estimator.fit(self.constraints.X, redY, **{self.sample_weight_name: redW})
+        # To do 1: add other resampling methods in fairlearn.reductions.resampling
+        # To do 2: let users pass their own resampling methods, and we orchestrate here
+        # now we only consider this resampling method：fixed_size_deterministic
+        else:
+            resampled_X, resampled_y = resampling.resample_fixed_size_deterministic(self.constraints.X, redY, redW)
```

Comment:
I think that this shouldn't happen by default - the resampling method should be an argument to the constructor, and we should throw an exception if there's no `sample_weight` argument and no resampler (this is sort of implicit in your To Dos, but even if there's only one way available, I don't think we should do it "behind the user's back").

Of course... what to do if there is a resampler, but sample weights are available in the estimator.... my preference would be to throw an exception there (since the user may be confused), but opinions may well differ on that.